libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -32,43 +34,68 @@ export class DefaultConfigurationSettingsConverter implements ConfigurationSetti
     >();
-    let featureFlagsConfigSettings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
-    let foundFeatureManagement = false;
-    let featureFlags: any = {};
-    let featureManagementKeyWord = "";
-    let enabledForKeyWord = "";
+
+    let featureFlagsConfigSettings = new Array<SetConfigurationSettingParam<MsFeatureFlagValue>>();
+    let foundMsFmSchema = false;
+    let foundLegacySchema = false;
+    let legacySchemaFeatureManagementKeyWord = "";
+
+    const featureFlagsDict: any = {};
 
     if (options.format == ConfigurationFormat.Properties && !options.skipFeatureFlags) {
       this.checkFeatureManagementExist(config);
     }
 
-    for (let i = 0; i < Constants.FeatureManagementKeyWords.length; i++) {
+    for (let i = 0; i < Constants.FeatureManagementKeyWords.length - 1; i++) {
       if (Constants.FeatureManagementKeyWords[i] in config) {
-        if (foundFeatureManagement) {
+        if (foundLegacySchema) {
           throw new ArgumentError(
-            "Unable to proceed because data contains multiple sections corresponding to Feature Management."
+            `Unable to proceed because data contains multiple sections corresponding to Feature Management. with the key, ${Constants.FeatureManagementKeyWords[i]}`
           );
         }
+        
+        foundLegacySchema = true;
+        legacySchemaFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[i];
+
+        const legacyFeatureManagementKey = legacySchemaFeatureManagementKeyWord as keyof object;
+        featureFlagsDict[legacyFeatureManagementKey] = config[legacyFeatureManagementKey];
+        delete config[legacyFeatureManagementKey];
+      }
+    }
+
+    const msFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[3];
+    if (msFeatureManagementKeyWord in config) {
+      if (foundLegacySchema &&
+        Object.keys(config[msFeatureManagementKeyWord as keyof object]).some(key => key !== Constants.FeatureFlagsKeyWord)) {
+        throw new ArgumentError(
+          `Unable to proceed because data contains an already defined legacy schema section with the key, ${legacySchemaFeatureManagementKeyWord}.`
+        );
+      }
+
+      if (Object.keys(config[msFeatureManagementKeyWord as keyof object]).includes(Constants.FeatureFlagsKeyWord)) {
+        foundMsFmSchema = true;
+      }
 
-        featureManagementKeyWord = Constants.FeatureManagementKeyWords[i];
-        enabledForKeyWord = Constants.EnabledForKeyWords[i];
-        const featureManagementKey = featureManagementKeyWord as keyof object;
-        featureFlags = config[featureManagementKey];
-        delete config[featureManagementKey];
-        foundFeatureManagement = true;
+      if (Object.keys(config[msFeatureManagementKeyWord as keyof object]).some(key => key !== Constants.FeatureFlagsKeyWord)){
+        foundLegacySchema = true;
+        legacySchemaFeatureManagementKeyWord = msFeatureManagementKeyWord;
```

Comment:
On line 85 do we need to delete the "feature_management" using the legacy schema from config ?